scripts/update-pr-desc.py

```diff
@@ -0,0 +1,165 @@
+    if resp.status_code == 404:
+        raise ValueError("Commit not found")
+
+    result = resp.json()
+    if not result:
+        raise ValueError("No PRs associated with this commit")
+    if len(result) > 1:
+        logger.warning("Multiple PRs found for commit. Using the first one.")
+
+    return result[0].get("html_url", None)
+
+
+def parse_pr_url(url: str) -> tuple:
+    """Parse the PR URL and return the repo and PR number."""
+    escaped_owner = re.escape(OWNER)
+    regex = re.compile(rf"github\.com\/{escaped_owner}\/([a-zA-Z-]+)\/pull\/(\d+)")
+    result = regex.search(url)
+
+    if not result or len(result.groups()) != 2:
+        raise ValueError("Invalid PR URL")
+
+    return result.groups()
+
+
+def create_new_description(body: Optional[str], commit_url: str, append: bool):
+    """Create a new description for the PR."""
+    if not body:
+        logger.warning("No description found. Creating a new one.")
+        body = DEFAULT_BODY
+        append = False
+
+    if commit_url in body:
+        logger.warning("Commit already in the PR's description")
+        return body
+
+    changelog_header = ""
+    if not append and CHANGELOG_HEADER not in body:
+        changelog_header = f"\n\n{CHANGELOG_HEADER}\n"
+    new_body = f"{body.rstrip()}{changelog_header}\n- {commit_url}"
+
+    return new_body
+
+
+def update_description(url: str, sha: str, token: str, append: bool = False):
+    """Update the PR's description with the commit URL."""
+    try:
+        repo, pr_number = parse_pr_url(url)
+        pr_body = get_pr_body(repo, pr_number, token)
+        commit_pr_url = get_commit_pr_url(sha, token)
+        if not commit_pr_url:
+            raise ValueError("No PR URL found for the commit")
+
+        new_desc = create_new_description(pr_body, commit_pr_url, append)
+        logger.info("New description:\n%s", new_desc)
+
+        update_pr_body(repo, pr_number, token, new_desc)
+        logger.info("Updated the PR's description successfully")
+    except Exception as e:
+        logger.error("Failed to update the description: %s", e)
+
+
+def main():
+    """Run the script."""
+    parser = argparse.ArgumentParser(
+        prog="Update PR description",
+        description="Updates the provided PR's description with the commit",
+    )
+    parser.add_argument("--url", required=True, help="URL of the PR to be updated")
+    parser.add_argument(
+        "--commit",
+        required=True,
+        help="The Commit SHA which will be included in the PR's description",
+    )
+    parser.add_argument(
+        "-a",
+        "--append",
+        default=False,
+        help="Whether to append to the existing list of commits or create it",
+        action="store_true",
+    )
+
+    args = parser.parse_args()
+    token = os.getenv("GITHUB_TOKEN")
+
+    if not token:
+        parser.error(
+            "Github token is required. Provide it with --token or set GITHUB_TOKEN environment variable."
```

Comment:
```suggestion
            "Github token is required. Provide it by setting the GITHUB_TOKEN environment variable."
```

I think this was missed when removing `--token` support